kimchi/src/folding/expressions.rs

```diff
@@ -500,3 +502,67 @@ pub fn folding_expression<C: FoldingConfig>(
     }
     (integrated, extended_witness_generator)
 }
+
+impl<F, Config: FoldingConfig> From<ConstantExprInner<F>> for FoldingCompatibleExprInner<Config>
+where
+    Config::Curve: AffineCurve<ScalarField = F>,
+    <Config as FoldingConfig>::Challenge: From<ChallengeTerm>,
+{
+    fn from(expr: ConstantExprInner<F>) -> Self {
+        match expr {
+            ConstantExprInner::Challenge(chal) => {
+                FoldingCompatibleExprInner::Challenge(chal.into())
+            }
+            ConstantExprInner::Constant(c) => match c {
+                ConstantTerm::Literal(f) => FoldingCompatibleExprInner::Constant(f),
+                _ => panic!("ConstantExprInner not supported in folding expressions"),
```

Comment:
it isn't completely unsupported, you can replace the other variants by the field element they represent and use `FoldingCompatibleExprInner::Constant(f)` for all. Maybe we can add an associated type to the config to allow more complex constant in expressions, but you would have to then resolve it in the environment, and I think it will end being even more work.

Reply:
Oooh I see, I will work a bit on it and if it ends up being a small change I will add one commit for it. Otherwise, open a follow up pr.